src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2871,29 +2902,57 @@ def add(attr):
                 "method": "pool.dataset.permission",
                 "params": ["tank/myuser", {
                     "user": "myuser",
+                    "acl": [],
                     "group": "wheel",
                     "mode": "755",
-                    "recursive": true,
+                    "options": {"recursive": true, "stripacl": true},
                 }]
             }
         """
         path = (await self._get_instance(id))['mountpoint']
         user = data.get('user', None)
         group = data.get('group', None)
+        uid = gid = -1
         mode = data.get('mode', None)
-        recursive = data.get('recursive', False)
-        acl = data['acl']
+        options = data.get('options', {})
+        acl = data.get('acl', [])
+
         verrors = ValidationErrors()
+        if user:
+            uid = (await self.middleware.call('dscache.get_uncached_user', user))['pw_uid']
```

Comment:
I believe you need some sort of validation here, catching KeyError if the user does not exist?